Fortran/MOD_MAIN.f90

```diff
@@ -247,7 +249,7 @@ SUBROUTINE dEBM_core(tempm, swdm, swd_TOAm, emissm, clcov, ppm, tmpSNH, lastyear
       c1cs = (epsi*epsa_cst*4.*bolz*(T0**3)+beta)
       c2oc = (-epsi+epsa_oct*epsi)*bolz*(T0**4)-residual
       c1oc = (epsi*epsa_oct*4.*bolz*(T0**3)+beta)
-      winkelns = asin(-c2cs/(1.0_WP-Ans)/(S0(month)*tau_cs))*180.0_WP/pi
+      winkelns = asin(min(0.0_WP,max(1.0_WP,-c2cs/(1.0_WP-Ans)/(S0(month)*tau_cs))))*180.0_WP/pi
```

Comment:
This is somewhat wrong... I think you want to make sure that the argument of the asine function will be in the range of [-1. 1.] . This then should rather be:

```suggestion
      winkelns = asin(max(-1.0_WP,min(1.0_WP,-c2cs/(1.0_WP-Ans)/(S0(month)*tau_cs))))*180.0_WP/pi
```